APCAppCore/APCAppCore/UI/Onboarding/SignIn/APCSignInViewController.m

```diff
@@ -343,87 +347,14 @@ - (void)taskViewController:(ORKTaskViewController *)taskViewController didFinish
-    }
-    
-    if (consentResult.signature.requiresName && (consentResult.signature.givenName && consentResult.signature.familyName))
-    {
-        APCUser *user = [self user];
-        user.consentSignatureName = [consentResult.signature.givenName stringByAppendingFormat:@" %@",consentResult.signature.familyName];
-        user.consentSignatureImage = UIImagePNGRepresentation(consentResult.signature.signatureImage);
-        
-        NSDateFormatter *dateFormatter = [NSDateFormatter new];
-        dateFormatter.dateFormat = consentResult.signature.signatureDateFormatString;
-        user.consentSignatureDate = [dateFormatter dateFromString:consentResult.signature.signatureDate];
-        
-        // extract the user's sharing choice
-        APCConsentTask *task = taskViewController.task;
-        ORKConsentSharingStep *sharingStep = task.sharingStep;
-        APCUserConsentSharingScope sharingScope = APCUserConsentSharingScopeNone;
-        
-        for (ORKStepResult* result in taskViewController.result.results) {
-            if ([result.identifier isEqualToString:sharingStep.identifier]) {
-                for (ORKChoiceQuestionResult *choice in result.results) {
-                    if ([choice isKindOfClass:[ORKChoiceQuestionResult class]]) {
-                        NSNumber *answer = [choice.choiceAnswers firstObject];
-                        if ([answer isKindOfClass:[NSNumber class]]) {
-                            if (0 == answer.integerValue) {
-                                sharingScope = APCUserConsentSharingScopeStudy;
-                            }
-                            else if (1 == answer.integerValue) {
-                                sharingScope = APCUserConsentSharingScopeAll;
-                            }
-                            else {
-                                APCLogDebug(@"Unknown sharing choice answer: %@", answer);
-                            }
-                        }
-                        else {
-                            APCLogDebug(@"Unknown sharing choice answer(s): %@", choice.choiceAnswers);
-                        }
-                    }
-                }
-                break;
-            }
-        }
-        
-        user.sharingScope = sharingScope;
-        
-        [self dismissViewControllerAnimated:YES completion:^
-         {
-             [((APCAppDelegate*)[UIApplication sharedApplication].delegate) dataSubstrate].currentUser.userConsented = YES;
-             
+    if ([[self onboardingManager] checkForConsentWithTaskViewController:taskViewController]) {
+        [self dismissViewControllerAnimated:YES completion:^{
              [self sendConsent];
          }];
     }
     else
     {
-        [taskViewController dismissViewControllerAnimated:YES completion:^
-         {
-             [[NSNotificationCenter defaultCenter] postNotificationName:APCUserDidDeclineConsentNotification object:nil];
```

Comment:
Who listens for this notification, and will removing it from here break anything? Or was this the only place it's posted and should the listen code be cleaned up?